docs/en/operations/settings/settings.md

```diff
@@ -4653,6 +4653,40 @@ Result:
 └─────────────┴───────────┘
 ```
 
+## script_line_number {#script_line_number}
```

Comment:
I put myself into the shoes of a reader and I got completely confused by the docs:

1. Lines 4657-4659 say that setting `script_line_number` specifies the value for field `system.query_log.script_line_number`. Does that mean the system table field will unconditionally (always) be this value? Makes no sense. Does this mean the setting configures the line number of the first query in a script, and each query increments the count by 1? Well that could make sense but such a setting would IMHO be overkill. A third option is that the setting is supposed to be provided with each statement (e.g. `SELECT 'hello' SETTINGS script_line_number = 1; SELECT 'world' SETTINGS script_line_number = 2`) such that the script has a way to pass additional metadata in the form of a "line number" into system.query_log? Please clarify.

2. In below example, you are querying `system.query_log` but you are confusingly _not_ using the setting.

Reply:
Fixed it a little bit, should be more clear right now